xlive/Blam/Engine/memory/data.cpp

```diff
@@ -0,0 +1,13 @@
+#include "stdafx.h"
+#include "data.h"
+
+void* datum_get(s_data_array* data, datum object_index)
```

Comment:
unnecessary but a suggestion for ease of casting?

```cpp
template<typename T>
T datum_get(...)
{
   ...
}

void* datum_get_unsafe(...)
{
   ...
}
```

Reply:
not really a fan of templates unless we __have__ to use them

Reply:
something like
```cpp
	template <typename T = void>
	inline T* get_tag_fast(datum tag)
	{
		return reinterpret_cast<T*>(&get_tag_data()[get_tag_instances()[DATUM_INDEX_TO_ABSOLUTE_INDEX(tag)].data_offset]);
	}
```

would make it default to void* and then if anyone wants it will can have hard casting

Reply:
It'll just end up increasing the compile time by creating the extra functions that have the return type set in the template